client/my-sites/backup/clone-flow/index.tsx

```diff
@@ -0,0 +1,448 @@
+			return cloneRoleCredentials.baseUrl;
+		}
+		if ( cloneRoleCredentials.site_url ) {
+			return cloneRoleCredentials.site_url;
+		}
+		return '';
+	};
+
+	const inProgressRewindStatus = useSelector( ( state ) => {
+		return getInProgressRewindStatus( state, siteId, backupPeriod );
+	} );
+	const { message, percent, currentEntry, status } = useSelector( ( state ) => {
+		return getRestoreProgress( state, siteId ) || ( {} as RestoreProgress );
+	} );
+
+	const CredSettings = {
+		action: 'edit',
+		host: 'generic',
+		role: 'alternate',
+	};
+
+	const requestClone = useCallback(
+		() =>
+			dispatch(
+				rewindClone( siteId, backupPeriod, { types: rewindConfig, roleName: CredSettings.role } )
+			),
+		[ dispatch, siteId, backupPeriod, rewindConfig, CredSettings.role ]
+	);
+	const onConfirm = useCallback( () => {
+		dispatch( setValidFrom( 'restore', Date.now() ) );
+		setUserHasRequestedRestore( true );
+		requestClone();
+	}, [ dispatch, setUserHasRequestedRestore, requestClone ] );
+
+	// Takes a destination as a vault role or blog id
+	const onSetDestination = useCallback(
+		( role: string ) => {
+			setCloneDestination( role );
+			setUserHasSetDestination( true );
+		},
+		[ setUserHasSetDestination, setCloneDestination ]
+	);
+
+	const onSetBackupPeriod = useCallback(
+		( period ) => {
+			// Grab the selected backup period / rewindId and set it
+			setBackupPeriod( period );
+			setUserHasSetBackupPeriod( true );
+			const displayDate = applySiteOffset( moment( parseFloat( period ) * 1000 ), {
+				gmtOffset,
+				timezone,
+			} ).format( 'LLL' );
+			setBackupDisplayDate( displayDate );
+		},
+		[ moment, gmtOffset, timezone ]
+	);
+
+	const loading = rewindState.state === 'uninitialized';
+	const { restoreId } = rewindState.rewind || {};
+
+	const disableClone = false;
+
+	const { data: logs } = useRewindableActivityLogQuery( siteId, {}, { enabled: !! siteId } );
+	const lastBackup = logs && logs.length > 0 ? logs[ 0 ] : undefined;
+
+	// Screen that allows user to add credentials for an alternate restore / clone
+	const renderSetDestination = () => (
+		<>
+			<CloneFlowStepProgress currentStep="destination" />
+			<h3 className="clone-flow__title">{ translate( 'Set a destination site' ) }</h3>
+			<p className="clone-flow__info">
+				{ translate( 'Input information about the site you want to clone to' ) }
+			</p>
+			<div className="clone-flow__advanced-credentials">
+				<AdvancedCredentials
+					action={ CredSettings.action }
+					host={ CredSettings.host }
+					role={ CredSettings.role }
+					onFinishCallback={ () => onSetDestination( CredSettings.role ) }
+					redirectOnFinish={ false }
+					goBackPath={ previousPath }
+				/>
+			</div>
+		</>
+	);
+
+	// Screen that allows user to select a backup point to clone
+	const renderSetBackupPeriod = () => (
+		<>
+			<CloneFlowStepProgress currentStep="clonePoint" />
+			<h3 className="clone-flow__title">{ translate( 'Select a Backup Point to Copy' ) }</h3>
```

Comment:
Maybe we could use caps only for the first word.

```suggestion
			<h3 className="clone-flow__title">{ translate( 'Select a backup point to copy' ) }</h3>
```